backend/tracim_backend/lib/cleanup/cleanup.py

```diff
@@ -0,0 +1,339 @@
+
+        logger.info(self, "delete content {}".format(content.content_id))
+        deleted_contents.append(content.content_id)
+        self.safe_delete(content)
+        return deleted_contents
+
+    def delete_workspace(self, workspace: Workspace) -> int:
+        """
+        Delete workspace and associated stuff
+        - all content in the workspace
+        - all roles
+        - all upload permission
+        - workspace agenda
+
+        :param workspace: workspace to delete
+        :return: workspace_id of deleted workspace
+        """
+
+        # INFO - G.M - 2019-12-11 - delete role on workspace
+        roles = self.session.query(UserRoleInWorkspace).filter(
+            UserRoleInWorkspace.workspace_id == workspace.workspace_id
+        )
+        for role in roles:
+            logger.info(
+                self,
+                "delete role for user {} of workspace {}".format(
+                    role.user_id, workspace.workspace_id
+                ),
+            )
+            self.safe_delete(role)
+
+        # INFO - G.M - 2019-12-11 - delete permissions on workspace
+        upload_permissions = self.session.query(UploadPermission).filter(
+            UploadPermission.workspace_id == workspace.workspace_id
+        )
+        for upload_permission in upload_permissions:
+            logger.info(
+                self,
+                "delete upload_permission {} of workspace".format(
+                    upload_permission.upload_permission_id, workspace.workspace_id
+                ),
+            )
+            self.safe_delete(upload_permission)
+
+            logger.info(self, "delete workspace {}".format(workspace.workspace_id))
+            self.safe_delete(workspace)
+            for content in workspace.contents:
+                self.delete_content(content)
+        workspace_id = workspace.workspace_id
+        self.safe_delete(workspace)
+        return workspace_id
+
+    def delete_workspace_agenda(self, workspace_id: int) -> typing.Optional[str]:
+        # INFO - G.M - 2019-12-11 - delete workspace agenda
+        if not self.app_config.CALDAV__ENABLED:
+            logger.info(
+                self,
+                "will not delete workspace {} agenda, because caldav feature is disabled".format(
+                    workspace_id
+                ),
+            )
+            return
+
+        agenda_dir = "{}{}{}{}".format(
+            self.app_config.CALDAV__RADICALE__STORAGE__FILESYSTEM_FOLDER,
+            "/collection-root",
+            self.app_config.CALDAV_RADICALE_WORKSPACE_PATH,
+            workspace_id,
+        )
+        logger.info(self, "delete workspace {} agenda dir : {}".format(workspace_id, agenda_dir))
+        self.safe_delete_dir(agenda_dir)
+        return agenda_dir
+
+    def delete_user_associated_data(self, user: User) -> None:
+        """deleted all stuff about user but user itself, his workspaces and content"""
```

Comment:
wording ? I d'ont understand

Reply:
Did you prefer "except" ?
```suggestion
        """deleted all stuff about user except user itself, his workspaces and user content"""
```